examples/datasets/causal_data_example.py

```
@@ -0,0 +1,489 @@
+    G.add_nodes_from(all_nodes)
+    
+    # Add edges with weights
+    for source, target, weight in zip(sources, targets, weights):
+        if weight > 0:  # Only add edges with positive weights
+            G.add_edge(source, target, weight=weight)
+    
+    # Create layout
+    try:
+        # Try hierarchical layout for causal graphs
+        pos = nx.spring_layout(G, k=2, iterations=50, seed=42)
+    except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)